Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -2442,6 +2446,68 @@ def to_context(self):
             }
             return ret_value
 
+    class CustomIndicator(Indicator):
+
+        def __init__(self, indicator_type, value, dbot_score, params, prefix_str):
+            """
+            :type indicator_type: ``Str``
+            :param indicator_type: type name of the indicator
+
+            :type value: ``Any``
+            :param value: Value of the indicator
+
+            :type dbot_score: ``DBotScore``
+            :param dbot_score: If custom indicator has a score then create and set a DBotScore object.
+
+            :type params: ``Dict(Str,Any)``
+            :param params: A dictionary containing all the param names and their values
+
+            :type prefix_str: ``Str``
+            :param prefix_str: Will be used as the context path prefix
+
+            :return: None
+            :rtype: ``None``
+            """
+            if hasattr(DBotScoreType, indicator_type.upper()):
+                raise ValueError('Creating a custom indicator type with an existing type name is not allowed')
+            if not value:
+                raise ValueError('value is mandatory for creating the indicator')
+            if not prefix_str:
+                raise ValueError('prefix_str is mandatory for creating the indicator')
+
+            self.CONTEXT_PATH = '{prefix_str}(val.value && val.value == obj.value)'.format(prefix_str=prefix_str)
+
+            self.value = value
+
+            if not isinstance(dbot_score, Common.DBotScore):
+                raise ValueError('dbot_score must be of type DBotScore')
+
+            self.dbot_score = dbot_score
+            self.indicator_type = indicator_type
+            self.params = params
+            INDICATOR_TYPE_TO_CONTEXT_KEY[indicator_type.lower()] = indicator_type.capitalize()
```

Comment:
why do you need this line?

Reply:
I guess you're refereeing to the last line, All the indicator types names are stored in that dictionary, and there are functions where a given indicator type is checked to exist in the that dicationary, so I needed to add the name of the custom indicator to that dicationary.

Reply:
I guess you're refereeing to the last line, All the indicator types names are stored in that dictionary, and there are functions where a given indicator type is checked to exist in the that dicationary, so I needed to add the name of the custom indicator to that dicationary.